### PR TITLE
[move] Unify & fix layout construction

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

This PR unifies ways to construct a type layout (non-annotated vs fully annotated). Because of copy-paste, the previous implementation of fully annotated layout construction did not account for node count.

 - Return type of layout construction APIs carries information if it has delayed fields or not, to make sure clients handle these cases, reducing risk of errors.
 - Single function to construct type layout (but still two functions for struct layouts for now).
 - Unified flags/counters under a common context which checks for exceeding the limit. 

### Test Plan

It is a bit tricky to add tests because loader and struct indices needs to be mocked, working on it:
- [ ] test counts
- [ ] test depths
- [ ] test identifier mappings are correctly propagated

### CODE REVIEW

File: third_party/move/move-vm/runtime/src/native_functions.rs [hunk 1]
```diff
@@ -149,19 +149,14 @@ impl<'a, 'b, 'c> NativeContext<'a, 'b, 'c> {
         self.resolver.loader().type_to_type_tag(ty)
     }
 
-    pub fn type_to_type_layout(&self, ty: &Type) -> PartialVMResult<MoveTypeLayout> {
+    pub fn type_to_type_layout(&self, ty: &Type) -> PartialVMResult<(MoveTypeLayout, bool)> {
```
Comment (runtian-zhou) [inline third_party/move/move-vm/runtime/src/native_functions.rs:152]: Can you add a comment for what this bool mean?

**Comment by gelash** [comment]
@georgemitenkov was there any issues with the approach here?

**Comment by georgemitenkov** [comment]
@gelash forgot about this one: I think we benefit from having a single way of constructing layouts, so that callers can decide what to do with bool flag for an aggregator. Though adding an invariant violation on dropping true value inside the conversion might be cleaner?

**Comment by gelash** [comment]
@georgemitenkov it might have been to save the traversal to extract delayed field IDs (was there another one?), and instead compute it during the construction itself? (and generate some high level format, such as with IDs)

**Pull request closed without merging**